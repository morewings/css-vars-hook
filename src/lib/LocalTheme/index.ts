export {useLocalTheme} from './useLocalTheme';
export type {LocalRootProps} from './LocalRoot';
