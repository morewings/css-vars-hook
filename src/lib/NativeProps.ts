import type {AriaAttributes, HTMLAttributes} from 'react';

export type DataAttributeKey = `data-${string}`;

export type DataAttributes = Record<DataAttributeKey, string>;

export type LibraryProps<TElement = HTMLDivElement> = AriaAttributes & {
    id?: string;
    /**
     * Set native ARIA role attribute
     * @see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles
     */
    role?: HTMLAttributes<TElement>['role'];
    /**
     * Specify additional CSS class. This allows you to use styled(Component)
     * or the css prop in styled-components or emotion.
     */
    className?: HTMLAttributes<TElement>['className'];
};
