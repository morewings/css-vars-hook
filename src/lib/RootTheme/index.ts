export {useRootTheme} from './useRootTheme';
export {RootThemeProvider} from './RootThemeProvider';
export type {ProviderProps} from './RootThemeProvider';
