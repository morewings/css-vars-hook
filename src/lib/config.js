export const ROOT_ID = 'root-theme';
