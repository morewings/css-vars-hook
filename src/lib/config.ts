/**
 * @public
 * Id attribute of the library root div.
 */
export const ROOT_ID = 'root-theme';
