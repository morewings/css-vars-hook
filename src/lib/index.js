export {default as Counter} from './Counter';
export {useCssTheme} from './useCssTheme';
