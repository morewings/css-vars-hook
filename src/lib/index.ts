export {useTheme} from './useTheme';
export {useVariable} from './useVariable';
export {useRootTheme, RootThemeProvider} from './RootTheme';
export type {ProviderProps} from './RootTheme';
export type {ThemeType} from './ThemeType';
