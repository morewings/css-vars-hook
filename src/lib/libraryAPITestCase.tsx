import type {FC} from 'react';
import {createRef} from 'react';
import {render} from '@testing-library/react';

/**
 * Test utility containing a set of tests to cover Library API features.
 * @see https://storybook.olxtech.pl/?path=/docs/experimental-library-api--page
 * @see LibraryProps
 * @example
 * // inside test file
 * libraryAPITestCase(
 *  // provide a Component to test
 *  Accordion,
 *  // Set name to label tests
 *  'Accordion',
 *  // Add props required to run component
 *  { content: itemSingleComponent }
 *  );
 */
export const libraryAPITestCase = <TProps,>(
    Component: FC<TProps>,
    name: string,
    defaultProps = {} as TProps
) => {
    describe(`${name} - Native API`, () => {
        const propValue = 'bar';

        it('renders provided `data-` attributes', () => {
            const name = 'foo';

            const {getByTestId} = render(
                <Component
                    {...defaultProps}
                    data-foo={propValue}
                    data-testid={propValue}
                    className={propValue}
                />
            );
            expect(getByTestId(propValue)).toHaveAttribute(`data-${name}`, propValue);
            expect(getByTestId(propValue)).toHaveAttribute('data-testid', propValue);
        });

        it.each(['id', 'role'])('renders native string props', propName => {
            const {getByTestId} = render(
                <Component
                    {...defaultProps}
                    data-testid={propValue}
                    {...{[propName]: propValue}}
                />
            );
            expect(getByTestId(propValue)).toHaveAttribute(propName, propValue);
        });

        it('renders `className` prop', () => {
            const {getByTestId} = render(
                <Component
                    {...defaultProps}
                    data-testid={propValue}
                    className={propValue}
                />
            );
            expect(getByTestId(propValue).getAttribute('class')).toMatch(propValue);
        });

        it('accepts ref as a callback', () => {
            const ref = jest.fn();
            render(<Component {...defaultProps} ref={ref} />);
            expect(ref.mock.calls[0][0]).toMatchSnapshot();
        });

        it('accepts ref as a Mutable Object', () => {
            const ref = createRef<HTMLInputElement>();
            render(<Component {...defaultProps} ref={ref} />);
            expect(ref.current).toMatchSnapshot();
        });
    });
};
