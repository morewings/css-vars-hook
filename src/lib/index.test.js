import {renderHook} from '@testing-library/react-hooks';
import {useVariable, useTheme} from 'lib';

describe('lib exports shape', () => {
  it('has useTheme', () => {
    expect(useTheme).toMatchSnapshot();
    const {result} = renderHook(() => useTheme({}));
    expect(result.current).toMatchSnapshot();
  });

  it('has useVariable', () => {
    expect(useVariable).toMatchSnapshot();
    const {result} = renderHook(() => useVariable('foo', 'bar'));
    expect(result.current).toMatchSnapshot();
  });
});
