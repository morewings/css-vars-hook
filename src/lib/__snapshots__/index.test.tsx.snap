// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib exports shape has useTheme 1`] = `[Function]`;

exports[`lib exports shape has useTheme 2`] = `
Object {
  "getTheme": [Function],
  "getVariable": [Function],
  "ref": Object {
    "current": null,
  },
  "removeVariable": [Function],
  "setRef": [Function],
  "setTheme": [Function],
  "setVariable": [Function],
  "style": Object {},
}
`;

exports[`lib exports shape has useVariable 1`] = `[Function]`;

exports[`lib exports shape has useVariable 2`] = `
Object {
  "getVariable": [Function],
  "ref": Object {
    "current": null,
  },
  "removeVariable": [Function],
  "setRef": [Function],
  "setVariable": [Function],
  "style": Object {
    "--foo": "bar",
  },
}
`;
